plugins/test/unit/plugins/importers/yum/test_purge.py

```diff
@@ -288,3 +290,21 @@ def test_retain_old(self, mock_remove_old_versions):
         purge.purge_unwanted_units(self.metadata_files, self.conduit, self.config)
 
         mock_remove_old_versions.assert_called_once_with(3, self.conduit)
+
+
+class RemoveUnitDuplicateNevra(TestPurgeBase):
+
+    @mock.patch.object(purge, 'RepoUnitAssociationManager', autospec=True)
+    @mock.patch.object(purge, 'UnitAssociationCriteria', autospec=True)
+    def test_remove_unit_duplicate_nerva(self, mock_criteria, mock_association):
+        unit_key = {'name': 'test-nevra', 'epoch': 0, 'version': 1, 'release': '23',
+                    'arch': 'noarch', 'checksum': '1234abc', 'checksumtype': 'sha256'}
+        type_id = 'rpm'
+        repo_id = 'test-repo'
+        mock_criteria.return_value = {'$and': [{'some': 'criteria'}]}
+
+        purge.remove_unit_duplicate_nevra(unit_key, type_id, repo_id)
+
+        mock_association.unassociate_by_criteria.assert_called_once_with(
+            repo_id,
+            mock_criteria.return_value, None, None)
```

Comment:
Alternatively, you could add assertions that the mock_criteria was called correctly. I'd prefer not mocking the criteria, but this would be an acceptable alternative. Either way, this function needs stronger assertions.